logger.go

```diff
@@ -8,33 +8,31 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	log *logrus.Entry
-)
-
-func setupLogger(logFile, logLevel string) {
-	logger := logrus.New()
-	writer, err := os.OpenFile(logFile, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0644)
+func setupLogger(logFile, logLevel string) (*logrus.Entry, error) {
+	writer, err := os.OpenFile(logFile, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0o644)
 	if err != nil {
-		fmt.Printf("cannot open log file: %s", err)
-		os.Exit(1)
+		return nil, fmt.Errorf("cannot open log file %q: %w", logFile, err)
 	}
 
+	logger := logrus.New()
+
 	logger.SetOutput(writer)
 	logger.SetFormatter(&logrus.JSONFormatter{
 		TimestampFormat:   time.RFC3339Nano,
 		DisableHTMLEscape: true,
 	})
 
-	log = logrus.NewEntry(logger)
+	logEntry := logrus.NewEntry(logger)
 
 	level, err := logrus.ParseLevel(logLevel)
 	if err != nil {
 		level = logrus.DebugLevel
 
-		log.WithField("given_level", logLevel).
+		logEntry.WithField("given_level", logLevel).
 			Warn("could not parse log level, defaulting to 'debug'")
 	}
 
-	logrus.SetLevel(level)
+	logger.SetLevel(level)
```

Comment:
this change fixes an otherwise-unnoticed bug where the logger would only ever log at info level, regardless of what had been specified!